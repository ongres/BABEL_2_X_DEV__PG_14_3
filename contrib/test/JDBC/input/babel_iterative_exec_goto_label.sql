GOTO mylabel2
mylabel1:
SELECT 'mylabel1'
mylabel2:
SELECT 'mylabel2'
GO

-- test unsupported GOTOs
-- goto forward into a try block
GOTO mylabel
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
END CATCH
GO

-- goto forward into a catch block
GOTO mylabel
BEGIN TRY
	SELECT 1; -- dummy
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO

-- goto backward into a try block
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
END CATCH
GOTO mylabel
GO

-- goto backward into a catch block
BEGIN TRY
	SELECT 1; -- dummy
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GOTO mylabel
GO

-- goto forward into a loop
DECLARE @i int
SET @i = 9
GOTO mylabel
WHILE (@i < 10)
	BEGIN
		mylabel:
	END
GO

-- goto backward into a loop
DECLARE @i int
SET @i = 9
WHILE (@i < 10)
	BEGIN
		mylabel:
	END
GOTO mylabel
GO

-- goto from try block to catch block
BEGIN TRY
	GOTO mylabel
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO

-- goto from catch block to try block
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
	GOTO mylabel
END CATCH
GO

-- goto upper catch block
BEGIN TRY
	BEGIN TRY
		GOTO mylabel
	END TRY
	BEGIN CATCH
	END CATCH
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO

-- GOTO and escape from TRY-CATCH block
BEGIN TRY
	GOTO mylabel
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 as X-- shall not be caught, test if context is cleaned
GO

BEGIN TRY
	SELECT 1/0 -- shall not be caught, test if context is cleaned
END TRY
BEGIN CATCH
	GOTO mylabel
END CATCH
mylabel:
SELECT 1/0
GO

BEGIN TRY
	BEGIN TRY
		GOTO mylabel
	END TRY
	BEGIN CATCH
	END CATCH
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 -- shall not be caught, test if context is cleaned
GO

BEGIN TRY
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		GOTO mylabel
	END CATCH
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 -- shall not be caught, test if context is cleaned
GO

-- expect duplicate label error
GOTO label
LABEL:
SELECT 'Upper Case'
label:
SELECT 'Lower Case'
GO

-- accept trailing semi-colon
GOTO label1;
SELECT 'Should be skipped'
label1:
SELECT 'Label1 OK!'
GOTO label2
SELECT 'Should be skipped'
label2:
SELECT 'Label2 OK!'
GO
