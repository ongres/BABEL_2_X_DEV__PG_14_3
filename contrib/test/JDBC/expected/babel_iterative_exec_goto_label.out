GOTO mylabel2
mylabel1:
SELECT 'mylabel1'
mylabel2:
SELECT 'mylabel2'
GO
~~START~~
varchar
mylabel2
~~END~~


-- test unsupported GOTOs
-- goto forward into a try block
GOTO mylabel
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
END CATCH
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto forward into a catch block
GOTO mylabel
BEGIN TRY
	SELECT 1; -- dummy
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto backward into a try block
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
END CATCH
GOTO mylabel
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto backward into a catch block
BEGIN TRY
	SELECT 1; -- dummy
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GOTO mylabel
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto forward into a loop
DECLARE @i int
SET @i = 9
GOTO mylabel
WHILE (@i < 10)
	BEGIN
		mylabel:
	END
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an while loop not supported, label mylabel)~~


-- goto backward into a loop
DECLARE @i int
SET @i = 9
WHILE (@i < 10)
	BEGIN
		mylabel:
	END
GOTO mylabel
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an while loop not supported, label mylabel)~~


-- goto from try block to catch block
BEGIN TRY
	GOTO mylabel
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto from catch block to try block
BEGIN TRY
	mylabel:
END TRY
BEGIN CATCH
	GOTO mylabel
END CATCH
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- goto upper catch block
BEGIN TRY
	BEGIN TRY
		GOTO mylabel
	END TRY
	BEGIN CATCH
	END CATCH
END TRY
BEGIN CATCH
	mylabel:
END CATCH
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: GOTO into an try catch block not supported, label mylabel)~~


-- GOTO and escape from TRY-CATCH block
BEGIN TRY
	GOTO mylabel
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 as X-- shall not be caught, test if context is cleaned
GO
~~ERROR (Code: 8134)~~

~~ERROR (Message: division by zero)~~


BEGIN TRY
	SELECT 1/0 -- shall not be caught, test if context is cleaned
END TRY
BEGIN CATCH
	GOTO mylabel
END CATCH
mylabel:
SELECT 1/0
GO
~~ERROR (Code: 8134)~~

~~ERROR (Message: division by zero)~~


BEGIN TRY
	BEGIN TRY
		GOTO mylabel
	END TRY
	BEGIN CATCH
	END CATCH
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 -- shall not be caught, test if context is cleaned
GO
~~ERROR (Code: 8134)~~

~~ERROR (Message: division by zero)~~


BEGIN TRY
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		GOTO mylabel
	END CATCH
END TRY
BEGIN CATCH
END CATCH
mylabel:
SELECT 1/0 -- shall not be caught, test if context is cleaned
GO
~~ERROR (Code: 8134)~~

~~ERROR (Message: division by zero)~~


-- expect duplicate label error
GOTO label
LABEL:
SELECT 'Upper Case'
label:
SELECT 'Lower Case'
GO
~~ERROR (Code: 132)~~

~~ERROR (Message: Label label not unique wihtin one procedure in line 5, previous defined in line 3)~~


-- accept trailing semi-colon
GOTO label1;
SELECT 'Should be skipped'
label1:
SELECT 'Label1 OK!'
GOTO label2
SELECT 'Should be skipped'
label2:
SELECT 'Label2 OK!'
GO
~~START~~
varchar
Label1 OK!
~~END~~

~~START~~
varchar
Label2 OK!
~~END~~

